pkg/certificate/reconciler.go

```diff
@@ -0,0 +1,35 @@
+package certificate
+
+import (
+	"context"
+	"time"
+)
+
+// CheckAndUpdate checks for a MRC to reach a desired condition and updates the status when the condition is met
+func (r *MRCReconciler) CheckAndUpdate(ctx context.Context, mrcClient MRCClient, checkInterval time.Duration) {
```

Comment:
We need a test for this; let me know if you need help setting the test up.

Reply:
Added a unit test!